### ISSUE TITLE
Invisible blocks when near CBMultipart blocks

### ISSUE THREAD

**Issue by arceste** [TheCBProject/CBMultipart#120]
I'm experiencing an issue where certain blocks are invisible until I look directly at them, or in the case of storage drawers, the contents do not appear at all on the front whenever CBMultipart blocks are anywhere nearby. This only occurs in multiplayer, singleplayer solves all woes.
[A temporary solution was found](https://github.com/simibubi/Above-and-Beyond/issues/128) for this issue in Create: Above and Beyond by force-loading the relevant chunk, though this isn't ideal as a permanent fix.
I'm not running Optifine, Magnesium, or anything that isn't in Create: Above and Beyond 1.2 by default.

**Comment by Xandaros** [comment]
The same issue was reported for CodeChickenLib here: https://github.com/TheCBProject/CodeChickenLib/issues/374
However, this does seem to be a CBMultipart bug, so I will continue using this issue instead.

We did some more investigating and it appears that any Project Red block will cause this bug, even if it's just a lamp. I don't know of any other MultiPart blocks in the *Create: Above and Beyond* modpack, so we didn't test any others.

To reproduce:
- Log into a server with the relevant mods
- Find a chunk in a region that has no chunks which are permanently loaded - ideally far away from spawn
- Place an Inverted Magenta Laser Lamp from Project Red in that chunk (or, presumably, any other MultiPart block)
- Leave the area. Make sure every chunk in the region is unloaded
- Go back. It might be necessary to use a teleport, we did not test normal movement

You will not get an IndexOutOfBounds exception. Other tile entities in the same chunk may or may not be affected by this. We assume it has to do with the order in which their data is sent.

As a kind of hotfix, we changed `codechicken.multipart.block.TileMultiPart.handleDescPacket` to this:
```java
public static void handleDescPacket(World world, BlockPos pos, MCDataInput packet) {
    List<TMultiPart> parts = new ArrayList<>();
    int nParts;
    try {
        nParts = packet.readUByte();
    } catch (IndexOutOfBoundsException e) {
        LogManager.getLogger().warn("Index out of bounds exception for block: {}, {}, {}", pos.getX(), pos.getY(), pos.getZ());
        return;
    }
    for (int i = 0; i < nParts; i++) {
        parts.add(MultiPartRegistries.readPart(packet));
    }
    if (parts.isEmpty()) return;

    TileEntity t = world.getBlockEntity(pos);
    TileMultiPart tile = MultiPartGenerator.INSTANCE.generateCompositeTile(t, parts, true);
    if (tile != t) {
        world.setBlockAndUpdate(pos, CBMultipartModContent.blockMultipart.defaultBlockState());
        MultiPartHelper.silentAddTile(world, pos, tile);
    }

    tile.loadParts(parts);
    tile.notifyTileChange();
    tile.markRender();
}
```

Presumably, this is not a proper fix for the problem, but it does contain the issue. The Project Red components actually seem to be working normally with this in pace - I assume this is because they constantly send updates and this problem only occurs on first load.
Regardless, containing exceptions here might be a good idea for the future, even if a proper fix is implemented.
